### PR TITLE
add an admin endpoint to remove libraries with inactive owners

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @gratimax

### CODE REVIEW

**Comment by camhashemi** [comment]
just merging for sake of time, won't call this until I get a thumbs up

File: kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala [hunk 1]
```diff
@@ -514,12 +517,18 @@ class LibraryRepoImpl @Inject() (
...
       val inIds = owners.map { _.id }.mkString("(", ",", ")")
-      val q = sql"""select owner_id, count(*) from library  where owner_id in #${inIds} and kind = 'user_created' and state = 'active' and visibility = 'published' group by owner_id"""
+      val q = sql"""select owner_id, count(*) from library where owner_id in #${inIds} and kind = 'user_created' and state = 'active' and visibility = 'published' group by owner_id"""
 
       val cnts = q.as[(Int, Int)].list.map { case (userId, count) => Id[User](userId) -> count }.toMap
       owners.map { user => user -> cnts.getOrElse(user, 0) }.toMap
     }
   }
+
+  def getLibrariesWithInactiveOwnerMemberships()(implicit session: RSession): Seq[(Id[Library], Id[User])] = {
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala:527]: The name is a bit misleading. You're looking for libraries with inactive owners. You're not really looking at their memberships.

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala [hunk 2]
```diff
@@ -363,4 +363,11 @@ class AdminLibraryController @Inject() (
     }
   }
 
+  def removeLibrariesWithInactiveOwner = AdminUserAction { implicit request =>
+    val libsAndOwners = db.readOnlyMaster { implicit session => libraryRepo.getLibrariesWithInactiveOwnerMemberships() }
+    implicit val context = HeimdalContext.empty
+    libsAndOwners.foreach { case (libId, userId) => libraryCommander.deleteLibrary(libId, userId) }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala:369]: This seems really dangerous. I'm hesitant to call methods via `userId` if the model referenced by the id is inactive.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala:369]: We can maybe have a method `libraryCommander.unsafeDeleteLibrary(libraryId: Id[Library])` that

File: kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala:527]: Yup

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala:369]: That's a good call. Also going to add asynchronous batching.

**Pull request merged**